deleting-compaction-strategy/src/main/java/com/protectwise/cassandra/util/DecimalSerializerMetaData.java

```diff
@@ -0,0 +1,62 @@
+package com.protectwise.cassandra.util;
+
+import com.protectwise.cassandra.retrospect.deletion.SerializableCellData;
+import org.apache.cassandra.config.ColumnDefinition;
+import org.apache.cassandra.db.Cell;
+import org.apache.cassandra.db.ColumnFamily;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.DecimalType;
+import org.apache.cassandra.db.marshal.LongType;
+import org.apache.cassandra.serializers.DecimalSerializer;
+import org.apache.cassandra.serializers.TypeSerializer;
+import org.apache.cassandra.utils.ByteBufferUtil;
+import org.codehaus.jackson.annotate.JsonTypeName;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Created by ayaz on 15/3/17.
+ */
+@JsonTypeName("DecimalSerializerMetaData")
+public class DecimalSerializerMetaData extends SerializerMetaData{
+    @Override
+    public TypeSerializer getSerializer() {
+        return DecimalSerializer.instance;
+    }
+
+    @Override
+    public SerializerMetaData getSerializerMetaData(TypeSerializer typeSerializer) {
+        if(typeSerializer instanceof DecimalSerializer) {
+            return this;
+        } else {
+            throw new RuntimeException("type serializer: " + typeSerializer.getClass().getName() + " is not compatible for class " + this.getClass().getName());
+        }
+    }
+
+    @Override
+    public SerializableCellData getSerializableCellData(Cell cell, ColumnFamily columnFamily) {
+        AbstractType<?> cellNameType = columnFamily.metadata().getValueValidator(cell.name());
+        try {
+            if(cellNameType instanceof DecimalType) {
+                SerializableCellData serializerCellData = new SerializableCellData();
```

Comment:
SerializableCellData serializerCellData = new SerializableCellData();
                //cell id is columnname
                String cellId = "";
                ColumnDefinition columnDefinition = columnFamily.metadata().getColumnDefinition(cell.name());
                cellId = String.join(CELL_ID_JOINER, columnDefinition.name.toString());
                serializerCellData.setCellId(cellId);
                serializerCellData.setValue(ByteBufferUtil.getArray(cell.value()));
                serializerCellData.setTimestamp(Long.valueOf(cell.timestamp()));
                return serializerCellData;


Above serialization logic is same in most of the primitive datatypes. Should we write a function in some util/super class?